packages/embark-namesystem/src/index.js

```diff
@@ -22,5 +25,15 @@ export default class Namesystem {
         }
       ]);
     });
+    embark.registerActionForEvent("pipeline:generateAll:before", this.addArtifactFile.bind(this));
+  }
+
+  addArtifactFile(_params, cb) {
+    this.events.request("pipeline:register", {
```

Comment:
Does this event just register to be executed later, or it executes straight away (generates on the spot I mean)
If it's the former, the action is not needed, we just need to register once

Reply:
it registers to be executed later, it's up to the pipeline (or who calls it) when to actually write to disk

Reply:
Ok, then we can just register once and not use the action. Otherwise, I think we will generate the same file multiple times (will be overwritten, but still, it could slow down)

Reply:
It wouldn't because if the same path+file is registered, it will override the previous one
https://github.com/embark-framework/embark/blob/master/packages/embark-pipeline/src/index.js#L19

I'll update the module documentation to refer to this